src/sentry/api/endpoints/project_group_index.py

```diff
@@ -112,24 +112,33 @@ def validate_repository(self, attrs, source):
         return attrs
 
     def validate(self, attrs):
```

Comment:
im not sure why i had to enforce the required checks here, but without them it was passing this validator successfully when I only sent the commit sha (and not the repo)